src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -1748,4 +1740,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
     return iface.isInstance(this);
   }
+
+  private <T> Map<String, Object> prepareMapWithValues(Object object, Class<T> type)
+      throws SQLException {
+    Map<String, Object> map;
```

Comment:
Such constructs are usually strange. Maybe we can inline it and return from both if branches?